app/js/controllers/settings.js

```diff
@@ -5,21 +5,39 @@
 export default class SettingsController extends Controller {
   constructor(options) {
     super(options);
+
+    this.merging = false;
+
+    this._refreshAddons = this._refreshAddons.bind(this);
   }
 
   open() {
     this.view.modal.open();
 
-    AddonService.getAddons(window.location.host).then((addons) => {
-      this.view.setAddons(addons);
+    this._refreshAddons();
+
+    ['install', 'uninstall', 'enabledstatechange'].forEach((e) => {
+      navigator.mozApps.mgmt.addEventListener(e, this._refreshAddons);
```

Comment:
We should file a follow-up to add a check that the affected app is 1. an add-on and 2. affects the current app. It's not important at this stage.